swayfmt/src/items/item_enum/mod.rs

```diff
@@ -19,94 +22,101 @@ impl Format for ItemEnum {
         formatted_code: &mut FormattedCode,
         formatter: &mut Formatter,
     ) -> Result<(), FormatterError> {
-        formatter
-            .shape
-            .code_line
-            .update_line_style(LineStyle::Multiline);
-        // If there is a visibility token add it to the formatted_code with a ` ` after it.
-        if let Some(visibility) = &self.visibility {
-            write!(formatted_code, "{} ", visibility.span().as_str())?;
-        }
-        // Add enum token and name
-        write!(formatted_code, "{} ", self.enum_token.span().as_str())?;
-        self.name.format(formatted_code, formatter)?;
-        // Format `GenericParams`, if any
-        if let Some(generics) = &self.generics {
-            generics.format(formatted_code, formatter)?;
-        }
+        formatter.with_shape(
+            Shape::from(
+                &formatter.shape,
+                Some(0), // In some cases we will want to update parts of Shape
+                Some(CodeLine::new(LineStyle::Multiline, ExprKind::Undetermined)),
+            ),
+            |formatter| -> Result<(), FormatterError> {
```

Comment:
One thing that might help to cut down on indentation a little is to first assign the new shape to a binding, e.g.

```rust
let new_shape = Shape::from(
    &formatter.shape,
    Some(0), // In some cases we will want to update parts of Shape
    Some(CodeLine::new(LineStyle::Multiline, ExprKind::Undetermined)),
);
formatter.with_shape(new_shape, |formatter| -> Result<(), FormatterError> {
```

Not important though, purely a style thing!

Reply:
One thing I plan to do after this PR is refactor `CodeLine` to have the `width` and `has_where_clause` fields since I think it actually belongs there instead of just on `Shape` itself. I think this will reduce the clutter and also remove the need to have the extra `Some(0)` 😄

Reply:
Nicce, just noticed you made these changes already - looking much better!

I think we can still reduce the indentation a little by first assigning the new shape to a variable like in my example above. E.g. with your new changes, this might look like:

```rust
        let new_shape = formatter
            .shape
            .with_code_line_from(LineStyle::Multiline, ExprKind::default());
        formatter.with_shape(new_shape, |formatter| -> Result<(), FormatterError> {
            // Code starts here.
```
This removes ~4 spaces of indentation over what we currently have e.g.
```rust
        formatter.with_shape(
            formatter
                .shape
                .with_code_line_from(LineStyle::Multiline, ExprKind::default()),
            |formatter| -> Result<(), FormatterError> {
                // Code starts here.
```
That said, this is a very minor nit - feel free to ignore this one if you prefer the current way!

Reply:
It's just my personal preference to avoid adding variables where we won't re-use or update them, but this isn't based on any logic, just how I like to see it. Is it better to have it behind a variable for any reason other than reducing indentation?

Reply:
People have different monitor sizes and may set up multiple windows / tabs side by side. If things are heavily indented then they have to keep scrolling from side to side to see the content. If indentation can be cleaned up by assignment to variables and parsing around those vars then that's generally preferred.